### PR TITLE
some improvements to distance, apply and fill methods in Raster*

### PR DESCRIPTION
**Pull request by sebastianbaginski**
mostly reducing count of function calls (property reads)
@marcusnaslund

### CODE REVIEW

File: source/draw/RasterMonochrome.ooc [hunk 1]
```diff
@@ -106,7 +115,7 @@ RasterMonochrome: class extends RasterPacked {
 						result += distance sqrt()
 					}
 				}
-			result /= (this size x squared + this size y squared as Float sqrt())
+			result /= (sizeX squared + sizeY squared as Float sqrt())
```
Comment (sebastianbaginski) [inline source/draw/RasterMonochrome.ooc]: I'm pretty sure this is a bug (should be more like `(x^2 + y^2) sqrt()`) but fixing it fails the RasterMonochrome `lossy ascii serialization` test case.

File: source/draw/RasterYuv420Semiplanar.ooc [hunk 2]
```diff
@@ -172,9 +174,9 @@ RasterYuv420Semiplanar: class extends RasterImage {
 					vSource += 2
 				}
 			}
-			yRow += this y stride
+			yRow += yStride
 			if (y % 2 == 1) {
```
Comment (marcusnaslund) [inline source/draw/RasterYuv420Semiplanar.ooc]: You can remove these braces while you're at it

File: source/draw/RasterMonochrome.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/draw/RasterMonochrome.ooc]: Oh definitely. Please create an issue

**Comment by marcusnaslund** [comment]
Neat. I will just ask for two things:

1) Benchmark this and see how much better the changes are.

2) You have many new helper variables which have the same name as the class member (`stride := this stride` etc.). These should be renamed (`thisStride := this stride`, as you have done in some places, perhaps).

File: source/draw/RasterMonochrome.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/draw/RasterMonochrome.ooc]: I did it: #1824

**Comment by sebastianbaginski** [comment]
Tested by running one of the tests in other project, which depends heavily on `distance`. Before changes:
> Total time: 10.12 s
> Total time: 8.91 s

after switching to this branch:
> Total time: 4.14 s
> Total time: 4.37 s

I will update the variable names.

**Comment by sebastianbaginski** [comment]
Complete test suite in the other project, before `91.80 s`, after `79.98 s` .

File: source/draw/RasterYuv420Semiplanar.ooc (hunk 2 above)
Comment (sebastianbaginski) [inline source/draw/RasterYuv420Semiplanar.ooc]: Updated.

**Comment by marcusnaslund** [comment]
Nice! :) I'm happy once the variable names have changed.

**Comment by sebastianbaginski** [comment]
Variable names are updated, check the latest commit.

File: source/draw/RasterUv.ooc [hunk 3]
```diff
@@ -61,14 +66,15 @@ RasterUv: class extends RasterPacked {
 		vSource := vRow
 		width := this size x
 		height := this size y
+		stride := this stride
```
Comment (marcusnaslund) [inline source/draw/RasterUv.ooc]: well well well

File: source/draw/RasterUv.ooc [hunk 4]
```diff
@@ -44,9 +44,14 @@ RasterUv: class extends RasterPacked {
 			uv referenceCount decrease()
 	}
 	fill: override func (color: ColorRgba) {
-		for (y in 0 .. this size y)
-			for (x in 0 .. this size x)
-				this[x, y] = ColorUv new(color r, color g)
+		sizeY := this size y
+		sizeX := this size x
+		buffer := this buffer pointer as ColorUv*
+		stride := this stride / this bytesPerPixel
```
Comment (marcusnaslund) [inline source/draw/RasterUv.ooc]: This isn't the same as `this stride` obviously, but it should still have some other name.

File: source/draw/RasterRgba.ooc [hunk 5]
```diff
@@ -29,15 +29,23 @@ RasterRgba: class extends RasterPacked {
...
 	}
 	copy: override func -> This { This new(this buffer copy(), this size, this stride) }
 	apply: override func ~rgb (action: Func(ColorRgb)) {
-		for (row in 0 .. this size y) {
-			source := this buffer pointer + row * this stride
-			for (pixel in 0 .. this size x) {
+		sizeX := this size x
+		sizeY := this size y
+		buffer := this buffer pointer
+		stride := this stride
```
Comment (marcusnaslund) [inline source/draw/RasterRgba.ooc]: well well well

Comment (sebastianbaginski) [inline source/draw/RasterRgba.ooc]: Just refresh your browser :dolphin:

File: source/draw/RasterRgb.ooc [hunk 6]
```diff
@@ -45,15 +45,24 @@ RasterRgb: class extends RasterPacked {
 			rgb referenceCount decrease()
 	}
 	fill: override func (color: ColorRgba) {
-		for (y in 0 .. this size y)
-			for (x in 0 .. this size x)
-				this[x, y] = color toRgb()
+		sizeX := this size x
+		sizeY := this size y
+		buffer := this buffer pointer as ColorRgb*
+		stride := this stride / this bytesPerPixel
```
Comment (marcusnaslund) [inline source/draw/RasterRgb.ooc]: well well well

Comment (marcusnaslund) [inline source/draw/RasterRgb.ooc]: (both lines 50 and 51)

Comment (sebastianbaginski) [inline source/draw/RasterRgb.ooc]: Just hit the F5 button, there is nothing out there.

**Comment by marcusnaslund** [comment]
Trying to trick me, eh? You should fear my power :zap:

**Comment by sebastianbaginski** [comment]
sorry :sob: 
anything more here ?

**Comment by marcusnaslund** [comment]
No, looks good no. But as punishment, I will wait an extra minute before merging this.

**Pull request merged**